strawberry/datetime_scalar.py

```diff
@@ -0,0 +1,29 @@
+import datetime
+
+import aniso8601
+
+from .custom_scalar import scalar
+
+
+def _serialize_isoformatted(value):
+    return value.isoformat()
+
+
+Date = scalar(
+    datetime.date,
+    description="Date (isoformat)",
+    serialize=_serialize_isoformatted,
+    parse_value=aniso8601.parse_date,
+)
+DateTime = scalar(
+    datetime.datetime,
+    description="Date with time (isoformat)",
+    serialize=_serialize_isoformatted,
+    parse_value=aniso8601.parse_datetime,
+)
+Time = scalar(
+    datetime.time,
+    description="Time (isoformat)",
+    serialize=_serialize_isoformatted,
+    parse_value=aniso8601.parse_time,
+)
```

Comment:
can we add some tests for these? :) also I'd put them inside `strawberry/types/datetime.py`

Reply:
do you think we need to add parse_literal as well? https://github.com/graphql-python/graphene/blob/master/graphene/types/datetime.py

Reply:
I'll add tests later this week.

I'm not sure about `parse_literal`. It is unclear to me under which conditions this method gets invoked (and I find neither the graphql-core nor the Graphene documentation very helpful on this). [There also seems to be a default implemention](https://github.com/graphql-python/graphql-core-next/blob/712ef8435bbc3066e73b69da96b09c13de945080/src/graphql/type/definition.py#L394), but not sure if that one is appropriate.

Reply:
So I think I finally understand `parse_literal` (or at least better than before): the GraphQL query is parsed into an AST, then `parse_literal` is used to convert the node representing the value of a scalar into a native Python object, and then `parse_value` converts that Python object to the final object that we want for the scalar.

The date/time objects will be serialized and deserialized as simple strings. Thus, the default `parse_literal` converting the `StringValueNode` from the AST to a Python string is sufficient because we then take that string in `parse_value` to convert it into the proper object. I also added a test verifying the parsing of literal date/time objects.

Looking at [Graphene](https://github.com/graphql-python/graphene/blob/master/graphene/types/scalars.py) the `parse_literal` could be used to ensure that the AST node if of the expected type. But the `parse_value` would also fail (with a reasonable exception) if we do not get a `StringValueNode`.

Reply:
That makes sense! Thanks for spending time on this!